addons/pos_restaurant/static/tests/tours/helpers/ProductScreenTourMethods.js

```diff
@@ -52,9 +52,41 @@ class DoExt extends Do {
             },
         ];
     }
+    clickOrderButton() {
+        return [
+            {
+                content: "click order button",
+                trigger: ".actionpad .submit-order",
+            },
+        ];
+    }
 }
 
 class CheckExt extends Check {
+    orderlinesIsNotToOrder() {
+        return [
+            {
+                content: "click order button",
+                trigger: ".actionpad .submit-order:not(.highlight)",
+                run: function () {
+                    if (document.querySelector(".orderlines .orderline.dirty")) {
+                        throw Error("The command lines must still be sent");
+                    } else {
+                        document.querySelector(".modal-dialog .cancel").click();
+                    }
+                },
```

Comment:
remove this. replace with 2 additional steps:
1. asserts that the orderline is not dirty. The trigger (selector) will be something like: `.orderlines .orderline:not(:has(.dirty))`.
2. asserts that the printing failed because no printer, so the dialog should be clicked. The trigger will be: `.modal-dialog .cancel`.

As much as possible, avoid putting logic in run method. Just create steps that will check presence (or absence if possible) of html elements.